### PR TITLE
feat: async account deletion & progress reset

### PR DESCRIPTION
**Pull request by luchaos**
... extracted from the much larger async double-write/metrics branch which will explain the queues' behaviour better.

Clean up site commands while at it.

Tests have been adjusted (thanks @Jamiras for putting those into place) but I wasn't able to verify _everything_ in time, please have another sanity check for those progress reset actions.

### CODE REVIEW

File: app/Site/Events/UserDeleted.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use App\Site\Models\User;
+use Illuminate\Broadcasting\InteractsWithSockets;
+use Illuminate\Broadcasting\PrivateChannel;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+class UserDeleted
+{
+    use Dispatchable;
+    use InteractsWithSockets;
```
Comment (Jamiras) [inline app/Site/Events/UserDeleted.php:14]: I'm still confused why we need `InteractsWithSockets`.

File: app/Platform/Listeners/ResetPlayerProgress.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    use Dispatchable;
+    use InteractsWithSockets;
+    use SerializesModels;
+
+    public function __construct(
+        public object $event,
+    ) {
+    }
+
+    public function handle(object $event): void
```
Comment (Jamiras) [inline app/Platform/Listeners/ResetPlayerProgress.php:25]: I get this error trying to run the `delete-overdue-accounts` command:
```
jamiras@ubuntu22 pr/1864:~/source/RAWeb$ php artisan ra:site:user:delete-overdue-accounts
   Illuminate\Contracts\Container\BindingResolutionException 

  Unresolvable dependency resolving [Parameter #0 [ <required> object $event ]] in class App\Platform\Listeners\ResetPlayerProgress
  
    at vendor/laravel/framework/src/Illuminate/Container/Container.php:1118
    1114▕     protected function unresolvablePrimitive(ReflectionParameter $parameter)
    1115▕     {
    1116▕         $message = "Unresolvable dependency resolving [$parameter] in class {$parameter->getDeclaringClass()->getName()}";
    1117▕ 
  ➜ 1118▕         throw new BindingResolutionException($message);
    1119▕     }
    1120▕ 
    1121▕     /**
    1122▕      * Register a new before resolving callback for all types.

      +13 vendor frames 
  14  app/Site/Actions/ClearAccountDataAction.php:80
      App\Site\Events\UserDeleted::dispatch()

  15  app/Site/Commands/DeleteOverdueUserAccounts.php:39
      App\Site\Actions\ClearAccountDataAction::execute()
```

File: app/Platform/Listeners/ResetPlayerProgress.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                $user = $event->user;
+                break;
+        }
+
+        if ($user === null) {
+            return;
+        }
+
+        // reset all achievements earned by the player
+        app()->make(ResetPlayerProgressAction::class)->execute($event->user);
```
[the remark this answers is not in the document]
Comment (luchaos) [reply app/Platform/Listeners/ResetPlayerProgress.php]: Addressed in https://github.com/RetroAchievements/RAWeb/pull/1864/commits/f5c891cbaa1381531181721ad14e45773ef302ce

File: app/Platform/Listeners/ResetPlayerProgress.php (hunk 2 above)
Comment (luchaos) [reply app/Platform/Listeners/ResetPlayerProgress.php:17]: Fixed in https://github.com/RetroAchievements/RAWeb/pull/1864/commits/f5c891cbaa1381531181721ad14e45773ef302ce

File: app/Platform/Listeners/ResetPlayerProgress.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+use Illuminate\Broadcasting\InteractsWithSockets;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Events\Dispatchable;
+use Illuminate\Queue\SerializesModels;
+
+class ResetPlayerProgress implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithSockets;
+    use SerializesModels;
```
[the remark this answers is not in the document]
Comment (luchaos) [reply app/Platform/Listeners/ResetPlayerProgress.php]: Good catch, those traits do not belong here. Bad copy.
Removed in https://github.com/RetroAchievements/RAWeb/pull/1864/commits/d4e01964534222b5af805aca123474331659d619

File: app/Site/Events/UserDeleted.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    use SerializesModels;
+
+    public function __construct(
+        public User $user
+    ) {
+    }
+
+    public function broadcastOn(): PrivateChannel
+    {
+        return new PrivateChannel('channel-name');
```
[the remark this answers is not in the document]
Comment (luchaos) [reply app/Site/Events/UserDeleted.php:24]: Absolutely, those channel names will receive proper names as soon as the websocket server is in place.

File: app/Site/Events/UserDeleted.php (hunk 1 above)
Comment (luchaos) [reply app/Site/Events/UserDeleted.php:14]: Allows events to be pushed to browser clients via websockets. A websocket server will put into place with an upcoming PR.
Those traits are from Laravel's default event stubs.

**Review by Jamiras — APPROVED** [review]

**Pull request merged**